### PR TITLE
feat: get theme list from server

### PR DESCRIPTION
**Pull request by zhangjing28**
## Description

* get ThemeList from the server
* handle transaction status 
* get reserveSbt fee
* start generate images after confirmed the theme

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [X] Test code changes manually, and describe your procedure in a comment on this PR
- [ ] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/pages/SBTPage/SBTContext/sbtPrivateWalletContext.tsx [hunk 1]
```diff
@@ -60,18 +81,104 @@ export const SBTPrivateContextProvider = ({
...
+      extensionSigner,
+      externalAccount.address
+    );
+
+    setTxStatus(TxStatus.processing(''));
+    try {
+      await reserveSbt.signAndSend(externalAccount.address, handleTxRes);
+    } catch (e: any) {
+      console.log('Failed to send a transaction', e);
+      setTxStatus(TxStatus.failed(e?.message ?? 'Transaction declined'));
```
Comment (Kevingislason) [inline src/pages/SBTPage/SBTContext/sbtPrivateWalletContext.tsx:165]: Unless you expect some particular error besides the transaction being declined, I would just do `TxStatus.failed('Transaction declined')`.

File: src/pages/SBTPage/SBTContext/sbtThemeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    const ret = await axios.post<{
+      model_id: string;
+    }>(url, data);
+    if (ret.status === 200 || ret.status === 201) {
+      setModelId(ret?.data?.model_id);
+    }
+  }, [
+    checkedThemeItems,
+    config.SBT_NODE_SERVICE,
+    externalAccount?.address,
```
Comment (Kevingislason) [inline src/pages/SBTPage/SBTContext/sbtThemeContext.tsx:78]: For these dependencies, it's preferable to just specify `config` and `externalAccount` for simplicity.

File: src/pages/SBTPage/SBTContext/sbtThemeContext.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      {children}
+    </SBTThemeContext.Provider>
+  );
+};
+
+export const useSBTTheme = () => {
+  const data = useContext(SBTThemeContext);
+  if (!data || !Object.keys(data)?.length) {
+    throw new Error(
+      'useTheme() can only be used inside of <useThemeContext />, please declare it at a higher level.'
```
Comment (Kevingislason) [inline src/pages/SBTPage/SBTContext/sbtThemeContext.tsx:104]: I think you mean `useSBTTheme` and `SBTThemeContextProvider`, right?

Comment (zhangjing28) [reply src/pages/SBTPage/SBTContext/sbtThemeContext.tsx:104]: Good eye! Will update that.

File: src/pages/SBTPage/components/ThemeCheckModal/index.tsx [hunk 4]
```diff
@@ -1,87 +1,94 @@
...
-import { useExternalAccount } from 'contexts/externalAccountContext';
 import { Step, useSBT } from 'pages/SBTPage/SBTContext';
 import { useSBTPrivateWallet } from 'pages/SBTPage/SBTContext/sbtPrivateWalletContext';
 import { useConfig } from 'contexts/configContext';
 import AssetType from 'types/AssetType';
+import { useTxStatus } from 'contexts/txStatusContext';
+import TxStatus from 'types/TxStatus';
+import { useSBTTheme } from 'pages/SBTPage/SBTContext/sbtThemeContext';
+
+const PRE_CHARGE_VALUE = 179.48;
```
[the remark this answers is not in the document]
Comment (zhangjing28) [reply src/pages/SBTPage/components/ThemeCheckModal/index.tsx:12]: It's the price of a SBT and i will change it to `PER_SBT_PRICE`. It's hard code by now, but will update it later

File: src/pages/SBTPage/components/ThemeCheckModal/index.tsx [hunk 5]
```diff
@@ -1,87 +1,94 @@
...
+            {reserveGasFee?.toString() ?? '-'} {nativeAsset?.baseTicker}
+          </span>
         </div>
         <div className="flex justify-between p-4">
           <p>Total</p>
           <div className="flex flex-col">
-            <span className="text-check">179.84 {nativeAsset?.baseTicker}</span>
-            <span className="text-white text-opacity-60">$30.9 USD</span>
+            <span className="text-check">
+              {totalValue} {nativeAsset?.baseTicker}
```
[the remark this answers is not in the document]
Comment (zhangjing28) [reply src/pages/SBTPage/components/ThemeCheckModal/index.tsx:89]: Happy to know that, will update

**Review by Kevingislason — APPROVED** [review]

**Pull request merged**